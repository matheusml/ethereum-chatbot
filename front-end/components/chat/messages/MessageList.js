import React from "react";
import { FlatList } from "react-native";
import PropTypes from "prop-types";

import MessageItem from "./MessageItem";

export default class MessageList extends React.Component {
  render() {
    return (
      <FlatList
        data={this.props.messages.reverse()}
        renderItem={({ item }) => <MessageItem item={item} />}
        keyExtractor={(item, index) => index}
        inverted
      />
    );
  }
}

MessageList.propTypes = {
  messages: PropTypes.array.isRequired
};
