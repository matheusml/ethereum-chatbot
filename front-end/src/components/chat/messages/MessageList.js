import React from "react";
import { ListView, ScrollView } from "react-native";
import PropTypes from "prop-types";

import MessageItem from "./MessageItem";

export default class MessageList extends React.Component {
  constructor(props) {
    super(props);

    this.renderItem = this.renderItem.bind(this);
  }

  renderItem(item) {
    return <MessageItem item={item} />;
  }

  render() {
    return (
      <ScrollView
        ref={ref => (this.scrollView = ref)}
        onContentSizeChange={(contentWidth, contentHeight) => {
          this.scrollView.scrollToEnd({ animated: true });
        }}
      >
        <ListView
          dataSource={this.props.messages}
          renderRow={this.renderItem}
          enableEmptySections={true}
        />
      </ScrollView>
    );
  }
}

MessageList.propTypes = {
  messages: PropTypes.object.isRequired
};
