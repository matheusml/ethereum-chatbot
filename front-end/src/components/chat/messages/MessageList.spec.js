import React from "react";
import { ListView } from "react-native";

import MessageList from "./MessageList";

import renderer from "react-test-renderer";

const props = {
  messages: new ListView.DataSource({
    rowHasChanged: (row1, row2) => row1 !== row2
  })
};

describe("MessageList", () => {
  it("should match snapshot", () => {
    const tree = renderer.create(<MessageList {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
