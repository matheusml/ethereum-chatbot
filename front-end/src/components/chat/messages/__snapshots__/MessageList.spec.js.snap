// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageList should match snapshot 1`] = `
<RCTScrollView
  onContentSizeChange={[Function]}
>
  <View>
    <RCTScrollView
      dataSource={
        ListViewDataSource {
          "items": "unknown",
        }
      }
      enableEmptySections={true}
      renderRow={[Function]}
      renderScrollComponent={[Function]}
    >
      <View />
    </RCTScrollView>
  </View>
</RCTScrollView>
`;
